src/app/domains/views/DomainDetails/DomainDetails.tsx

```diff
@@ -23,6 +26,9 @@ const DomainDetails = (): JSX.Element => {
     domainsSelectors.getById(state, Number(id))
   );
   const domainsLoading = useSelector(domainsSelectors.loading);
+  const [formOpen, setFormOpen] = useState<"DeleteDomain" | "AddRecord" | null>(
+    null
+  );
```

Comment:
This state handler needs to be integrated with sidePanelContent. Notice how pressing esc no longer works to close the side panel.

Take a look how that's implemented in `src/app/domains/views/DomainsList/constants.ts`. Then its type needs to be added to `SidePanelContent` in `src/app/base/side-panel-context.tsx`